src/components/main/ItemsToolbar.tsx

```diff
@@ -7,8 +7,8 @@ type Props = {
 
 const ItemsToolbar = ({ title, headerElements }: Props): JSX.Element => (
   <Stack direction="row" justifyContent="space-between" spacing={1}>
-    <Box>
-      <Typography variant="h4" noWrap>
+    <Box sx={{ maxWidth: '50%' }}>
```

Comment:
Maybe you can use https://mui.com/material-ui/react-grid/ for responsiveness? Or https://mui.com/material-ui/react-stack/ ?